pysteps/blending/steps.py

```diff
@@ -644,8 +644,30 @@ def forecast(
     else:
         # 2.3.3 If zero_precip_radar, make sure that precip_cascade does not contain
         # only nans or infs. If so, fill it with the zero value.
-        if zero_precip_radar:
-            precip_cascade[~np.isfinite(precip_cascade)] = np.nanmin(precip)
+        if np.any(precip_cascade[np.isfinite(precip_cascade)]):
+            precip_cascade[~np.isfinite(precip_cascade)] = np.nanmin(precip_cascade)
+        else:
+            done = False
```

Comment:
Is ``done`` set to ``True`` at some point in the code?

Reply:
Oops, forgot haha